src/jit/lir.h

```diff
@@ -72,36 +73,25 @@ class LIR final
         unsigned ReplaceWithLclVar(Compiler* compiler, unsigned blockWeight, unsigned lclNum = BAD_VAR_NUM);
     };
 
-    //------------------------------------------------------------------------
-    // LIR::Range: Represents a contiguous range of LIR nodes. Provides a
-    //             variety of utilites that modify the LIR contained in the
-    //             range.
-    //
-    class Range
+    class ReadOnlyRange
     {
         friend class LIR;
-        friend class SimpleRange;
+        friend class Range;
+        friend struct BasicBlock;
 
```

Comment:
The friend classes are deliberate: due to the possibility of object slicing, it's not straightforward to safely inherit from `LIR::Range`. I've attempted to restrict the possibility of error as much as possible by limiting the number of types that can subclass `LIR::Range` as well as making the type non-copyable.